lib/handlers/shared.ts

```diff
@@ -1,20 +1,69 @@
-import { ChainId, Currency, Percent } from '@uniswap/sdk-core'
+import { ChainId, Currency, CurrencyAmount, Fraction, Percent, WETH9 } from '@uniswap/sdk-core'
 import {
   AlphaRouterConfig,
   ITokenListProvider,
   ITokenProvider,
   MapWithLowerCaseKey,
   NATIVE_NAMES_BY_ID,
   nativeOnChain,
-  ProtocolPoolSelection,
+  ProtocolPoolSelection, USDC_MAINNET
 } from '@uniswap/smart-order-router'
 import Logger from 'bunyan'
+import JSBI from 'jsbi'
 
 export const SECONDS_PER_BLOCK_BY_CHAIN_ID: { [chainId in ChainId]?: number } = {
   [ChainId.MAINNET]: 30,
 }
 
-export const DEFAULT_ROUTING_CONFIG_BY_CHAIN = (chainId: ChainId): AlphaRouterConfig => {
+export const DEFAULT_ROUTING_CONFIG_BY_CHAIN = (chainId: ChainId, currencyIn: Currency, currencyOut: Currency, type: string, amountRaw: string, log: Logger): AlphaRouterConfig => {
```

Comment:
I woudl definitely get the distribution percent logic out into a new function

Reply:
Sure, separated into `DISTRIBUTION_PERCENT_CONFIG`